### PR TITLE
Fix remaining fast up to date check issues

### PR DESCRIPTION
**Pull request by jasonmalinowski**
With these workarounds, I'm able to do an incremental F6 in Visual Studio and it says everything is up to date. 

#12461 and #12458 have been filed tracking the two portable flavor issues I ran into. Internal bugs 239691 and 239697 have been filed on the two other native project system bugs.

*Review*: @TyOverby, @rchande, @jaredpar, @dotnet/roslyn-infrastructure, @tmat

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@dotnet-bot test mac please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot test vsi please.

**Comment by jasonmalinowski** [comment]
And this fixes #10847.

**Comment by Pilchie** [comment]
:+1:

**Comment by jasonmalinowski** [comment]
Actually ugh @rchande it looks like the netstandard hack might not work for Linux/Mac packages. Any other ideas?

**Comment by rchande** [comment]
I can't think of anything else...

**Comment by rchande** [comment]
:+1:

**Comment by jasonmalinowski** [comment]
Solution was to use netstandard1.6. I picked 1.3 arbitrarily, but obviously the 1.3 packages don't carry implementations along.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test mac please, test vsi please. (Retesting after I pushed.)

**Comment by CyrusNajmabadi** [comment]
:+1:

File: src/Interactive/CsiCore/project.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ﻿{
   "dependencies": { },
   "frameworks": {
+    "netstandard1.6": {
+      "imports": "portable-net452"
+    },
```
Comment (jaredpar) [inline src/Interactive/CsiCore/project.json]: Why were the netstandard 1.6 imports necessary?

Comment (tmat) [inline src/Interactive/CsiCore/project.json]: I'd like to know too.

Comment (jasonmalinowski) [inline src/Interactive/CsiCore/project.json]: It's working around #12458. There's a bug in the portable library tooling that is looking for something targeting netstandard to enable the proper compatibility checks. This is a hack to trick the project system into taking the right path, without actually affecting the build.

Comment (tmat) [inline src/Interactive/CsiCore/project.json]: And if .json was .xml we could have added a comment saying so. Sigh.

Comment (jaredpar) [inline src/Interactive/CsiCore/project.json]: I was just about to say "well add a comment" until I read @tmat comment 😢

Comment (jasonmalinowski) [inline src/Interactive/CsiCore/project.json]: Chatting with NuGet team, since they use Newtonsoft.Json comments should work here. I'll update this quick.

Comment (jasonmalinowski) [inline src/Interactive/CsiCore/project.json]: Updated.

**Pull request merged**